Wave/Wave/SingleSelectionBase.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  SingleSelectionBase.swift
+//  Wave
+//
+//  Created by Jimmy McDermott on 7/12/16.
+//  Copyright © 2016 BlueLine Labs. All rights reserved.
+//
+
+import Foundation
+
+//TODO: Make sure this supports only objects that can be serialized
+public class SingleSelectionBase<T: Any>: SectionType, SettingType {
+    
+    public var possibleValues: [T]
+    public var enableCustom = false
+    public var name: String
+    internal var settings = [RowType]()
+    private var selectedOption: Int?
+    internal var key: String
+    public var sectionDelegate: WaveDelegate?
+    private let inputCellIdentifier = "inputWaveCell"
+    private let displayCellIdentifier = "waveCell"
+    private var delegateProxy: UITextFieldDelegateProxy?
+    
+    public init(possibleValues: [T], enableCustom: Bool, name: String, key: String) {
+        self.possibleValues = possibleValues
+        self.enableCustom = enableCustom
+        self.name = name
+        self.key = key
+        
+        for value in possibleValues {
+            settings.append(StringSetting(placeholder: nil, defaultValue: nil, key: key, value: String(value), title: nil))
+        }
+        
+        if enableCustom {
+            settings.append(StringSetting(placeholder: nil, defaultValue: nil, key: "\(key)_customOption", value: "Custom Option", title: "Custom Option"))
+            
+            guard let customOptions = WaveKeeper.sharedInstance.getValueWithKey("\(key)_customOption") as? [T] else {
+                return
```

Comment:
If you return here, the selected option won't be selected. Be careful with guards and returns. This is a good place to just use `if let`.